apps/web/app/modules/Profile/components/ProfileActionButtons.tsx

```diff
@@ -0,0 +1,76 @@
+import { useTranslation } from "react-i18next";
+import { match } from "ts-pattern";
+
+import { Icon } from "~/components/Icon";
+import { Button } from "~/components/ui/button";
+
+import type { UseFormHandleSubmit, UseFormReset } from "react-hook-form";
+import type { UpdateUserProfileBody } from "~/api/generated-api";
+
+type ProfileActionButtonsProps = {
+  isEditing: boolean;
+  isProfileOwner: boolean;
+  toggleEditing: () => void;
+  copyLinkToClipboard: () => void;
+  handleSubmit: UseFormHandleSubmit<UpdateUserProfileBody, undefined>;
+  onSubmit: (data: Partial<UpdateUserProfileBody>) => void;
+  reset: UseFormReset<UpdateUserProfileBody>;
+};
+
+export const ProfileActionButtons = ({
+  isEditing,
+  isProfileOwner,
+  toggleEditing,
+  copyLinkToClipboard,
+  handleSubmit,
+  onSubmit,
+  reset,
+}: ProfileActionButtonsProps) => {
+  const { t } = useTranslation();
+
+  return match({ isEditing, isProfileOwner })
```

Comment:
you could create separate action buttons components and render them here, it would be better descriptive.